client/src/components/ProductDetails/index.jsx

```diff
@@ -21,6 +21,12 @@ const ProductDetails = ({ productId }) => {
     }
   }, [productId]);
 
+  useEffect(() => {
```

Comment:
repeated?

Reply:
I intended to use the first useEffect to watch productId changes and fetch product information from the API. 
The second product detail useEffect to watch the productStyles state to set a default selectedStyle if one isn't selected.

Though thanks for pointing this out, the selected style won't change if the product changes with my current setup. Addressed with PR #26 (ticket #100)